dashboard/app/controllers/data_docs_controller.rb

```diff
@@ -50,6 +55,17 @@ def update
     render json: @data_doc.serialize.to_json
   end
 
+  # DELETE /data_docs/:key
+  def destroy
+    begin
+      @data_doc.remove_serialization
+      @data_doc.destroy!
+      render(status: :ok, plain: "Destroyed #{@data_doc.key}")
+    rescue
+      render(status: :not_acceptable, plain: @data_doc.errors)
```

Comment:
> > > Oh for now that's all it's meant to show! The frontend will be in a separate PR, I'm just doing what I did with the [edit data doc PR](https://github.com/code-dot-org/code-dot-org/pull/48243) where there's just some text saying this is the right page. I could add a list element like the index page to temporarily show all created docs until the frontend PR is merged if that sounds good?
> > 
> > 
> > Oh I see - it does say it is the backend PR! this approach seems fine for now, maybe we can discuss the merits of this approach in standup sometime.
> 
> Yes that sounds good! I'd love some suggestions on how to best approach this!

Ditto –– I like vertical slicing *and* I'm not sure how to always do that in cases like these when PRs get big.

Reply:
(deleted my last comment because it was in the wrong place)
Good call re `:not_found` here! I like that approach (not rescuing errors in the destroy method)